Author: Upstream
Description: Report and fix by Mutsuhito Iikura. On finding a full buffer during
 matching the sliding window mechanism slides too far, truncating
 the whole buffer and preventing matches across the boundary. Fix
 is shortening the slide distance (slide only one 1/3).
Last-Modified: Fri, 23 Oct 2015 11:07:19 +0300
Bug: http://sourceforge.net/p/expect/bugs/76/

--- a/expect.c
+++ b/expect.c
@@ -979,7 +979,7 @@
 	/* condition is (size >= max*2/3) <=> (size*3 >= max*2) */
 	if (((expSizeGet(esPtr)*3) >= (esPtr->input.max*2)) && (numchars > 0)) {
 	o->e = e;
-	    o->matchlen = numchars;
+	    o->matchlen = numchars/3;
 	    o->matchbuf = str;
 	o->esPtr = esPtr;
 	expDiagLogU(yes);
